### PR TITLE
Made wavefunction equality obsolete to improve test consistency

### PR DESCRIPTION
**Pull request by clavigne**
Original tests had a lot of assert wf1 == wf2, where both wf1 and wf2 are QubitWaveFunction objects. Equality was tested in rather ad-hoc fashion, with a tolerance of 1e-6 elementwise for each wavefunction component. I got rid of == altogether here because it's somewhat ill-defined and replaced it instead with a isclose() function defined as

<psi1 | psi2> == sqrt(<psi1|psi1>) * sqrt(<psi2 | psi2>)

basically checking if the (normalized) overlap between the two wavefunctions is unity. This should be much more robust to numerical errors, while also allowing rtol and atol arguments. Furthermore, deprecating == means that we force the user to think about what they mean when they test whether two wavefunctions are "equal," which I think is a good thing.

### CODE REVIEW

**Comment by kottmanj** [comment]
Think some tests still depend on the good old == operator

**Comment by clavigne** [comment]
yeah that's why I set it as draft: to check where it would fail without having to install all the CI packages on my own machine 🙃

**Comment by clavigne** [comment]
Ok! appears to work. Should make CI more reliable.
